.github/workflows/publish.yml

```diff
@@ -10,23 +10,25 @@ permissions:
 jobs:
   publish:
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        python-version: [ "3.10" ]
 
     steps:
       - uses: actions/checkout@v3
 
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v2
+      - name: Set up Python
+        uses: actions/setup-python@v4
         with:
-          python-version: ${{ matrix.python-version }}
+          python-version: "3.10"
 
-      - name: Install dependencies
-        run: |
-          pip install -U pip poetry
+      - name: Install Poetry
+        uses: abatilo/actions-poetry@v2
+        with:
+          poetry-version: "1.2.0"
```

Comment:
This was a tiny command to run, what is the benefit of using a Github action instead? I'm just concerned about using third party Github actions if I don't have to.

Reply:
I am going to revert it for now (apart from pinning poetry version), as I realized that this github action is quite useless.

In general, I recommend looking into better ways of installing poetry even in the CI env as per their recommendations: https://python-poetry.org/docs/#ci-recommendations
It is recommended to install poetry in a dedicated venv as it can lead to some hard to debug situations when your deps clash with poetry's deps.